chore: add linter to the project

vcapretz opened the pull request:
I used xo for that, which is supposed to run with almost none configs,
but some changes were necessary to this project

- the indent rule is off because of a eslint bug:
https://github.com/eslint/eslint/issues/9779

I added the `.editorconfig` to help us formatting files 

closes #23

vcapretz committed:
chore: add linter to the project

I used xo for that, which is supposed to run with almost none configs,
but some changes were necessary to this project

- the indent rule is off because of a eslint bug:
https://github.com/eslint/eslint/issues/9779

lflimeira commented on test/acceptance/index.js:
```diff
@@ -1,22 +1,25 @@
-    })
-  })
-})
+					const expected = 'Plataforma training-center';
```
@vcapretz acho que deu ruim aqui.

lflimeira commented on daos/areasDAO.js:
```diff
@@ -1,34 +0,0 @@
-const models = require("../models/");
```
Você removeu esse arquivo?

vcapretz replied:
apenas renomeei, porque uma sugestão do eslint default é arquivos em snake-case

lflimeira replied:
Boaaa =)

lflimeira commented on test/acceptance/areas.js:
```diff
@@ -1,51 +1,51 @@
+chai.use(chaiHttp);
+
+describe('Route: Areas', () => {
+	it('GET areas/ - retorna lista de areas vazia', done => {
```
Aqui está utilizando tab, e em outros arquivos utiliza espaço.

lflimeira commented:
Eu estava pensando em usar o da Airbnb https://www.npmjs.com/package/eslint-config-airbnb-base o que vc acha?

vcapretz commented:
vou trocar entao e tirar o xo, colocar o eslint padrão e extender airbnb haha

vcapretz committed:
chore: update linter to use eslint

we should use eslint with the airbnb styleguide
it was used the `--fix` option to fix most of the things

vcapretz commented:
@lflimeira done! troquei o linter e as regras dele pra um mais padrão a lá Airbnb
